pypsa/optimization/abstract.py

```diff
@@ -161,6 +161,9 @@ def discretize_branch_components(
                 f"Optimization failed with status {status} and termination "
                 f"{termination_condition}"
             )
+            labels = n.model.compute_infeasibilities()
+            logger.info(f"Labels:\n{labels}")
+            n.model.print_infeasibilities()
```

Comment:
the idea is nice, however the `print_infeasiblities` only works with gurobi so far.

Reply:
this would be very useful for ariadne. maybe, we could add and if statement that checks if the solver is gurobi?